project-koku/koku
# [COST-4231] - handle empty OCP daily file

## Jira Ticket

[COST-4231](https://issues.redhat.com/browse/COST-4231)

## Description

This change will handle empty CSV in daily report payloads. This will move the empty file into S3 (the csv archive), but will not write a parquet file (since there is no data to write to parquet).

## Testing
`IN EPHEMERAL`
1. create OCP source with cluster-id: `test-001`:
```
$ oc port-forward service/sources-api-svc 3000:8000

$ curl --location 'http://localhost:3000/api/sources/v3.1/bulk_create' \
--header 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IjEwMDAxIiwib3JnX2lkIjoiMTIzNDU2NyIsInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6InVzZXJfZGV2IiwiZW1haWwiOiJ1c2VyX2RldkBmb28uY29tIiwiaXNfb3JnX2FkbWluIjp0cnVlLCJhY2Nlc3MiOnsgfX19LCJlbnRpdGxlbWVudHMiOnsiY29zdF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfX19Cg==' \
--header 'x-rh-sources-psk: thisMustBeEphemeralOrMinikube' \
--header 'Content-Type: application/json' \
--data '{
    "sources": [{
        "name":"username - ocp",
        "source_ref": "test-001",
        "source_type_name": "openshift"
    }],
    "endpoints":[],
    "authentications":[],
    "applications":[{
        "application_type_name":"cost-management",
        "source_name": "username - ocp"
    }]
}'
```
2. Ingest the attached payload. (This payload contains an empty storage report.):
```
$ oc port-forward service/ingress-service 20000:8000

$ curl -vvvv -F "file=@be3a9bff434641ae8a521c662e8d68e8.gz ;type=application/vnd.redhat.hccm.tar+tgz" -H "x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6IjEwMDAxIiwib3JnX2lkIjoiMTIzNDU2NyIsInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6InVzZXJfZGV2IiwiZW1haWwiOiJ1c2VyX2RldkBmb28uY29tIiwiaXNfb3JnX2FkbWluIjp0cnVlLCJhY2Nlc3MiOnsgfX19LCJlbnRpdGxlbWVudHMiOnsiY29zdF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfX19Cg==" http://localhost:20000/api/ingress/v1/upload
```
3. See ingestion succeed. There should only be one `warning` observed:
```
...
File "pyarrow/error.pxi", line 113, in pyarrow.lib.check_status
FileNotFoundError: [Errno 2] Failed to open local file ''. Detail: [errno 2] No such file or directory
[2023-09-27 14:35:31,626] WARNING None 1 {'message': 'failed to convert files to parquet', 'tracing_id': 'be3a9bff-4346-41ae-8a52-1c662e8d68e8', 'account': 'org1234567', 'provider_uuid': '73ebbe75-691f-4e58-bccf-141297ef87f0', 'provider_type': 'OCP', 'file_list': [PosixPath('/tmp/tmpl1fin1tr/insights_local/test-001/20230901-20231001/be3a9bff-4346-41ae-8a52-1c662e8d68e8_openshift_report.1.csv')]}
```
This warning is due to the empty storage report. This is an acceptable warning because the code is designed to skip storage processing if it does not exist.

## Notes

...
[be3a9bff434641ae8a521c662e8d68e8.gz](https://github.com/project-koku/koku/files/12739556/be3a9bff434641ae8a521c662e8d68e8.gz)
